### ISSUE TITLE
Importing existing management group at level_1 fails

### ISSUE THREAD

**Issue by adhodgson1** [Azure/terraform-azurerm-caf-enterprise-scale#613]
### Community Note
- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions
Terraform v1.3.6
+ provider registry.terraform.io/hashicorp/azurerm v3.43.0
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/hashicorp/time v0.9.1
Module version 3.1.2

### Description
I want to use this module to manage some additional management groups at level_1 in the hierarchy with custom_landing_zones. The Terraform plan completes and wants to add the group, but as the group already exists I want to import it into Terraform. My reasoning behind this is I want to use the capability of this module to assign policies etc to this level_1 MG.

#### Steps to Reproduce
main.tf:
```hcl
data "azurerm_client_config" "core" {}
module "enterprise_scale" {
  source  = "Azure/caf-enterprise-scale/azurerm"
  version = "3.1.2"

  providers = {
    azurerm              = azurerm
    azurerm.connectivity = azurerm
    azurerm.management   = azurerm
  }

  root_parent_id                  = data.azurerm_client_config.core.tenant_id
  deploy_core_landing_zones       = false
  disable_telemetry               = true
  custom_landing_zones            = {
    TestGroup = {
      display_name               = "Test Group"
      parent_management_group_id = ""
      subscription_ids           = []
      archetype_config = {
        archetype_id   = "default_empty"
        parameters     = {}
        access_control = {}
      }
    }
  }
}
```

Terraform plan:
```script
data.azurerm_client_config.core: Reading...
data.azurerm_client_config.core: Read complete after 0s [id=[...]]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # module.enterprise_scale.azurerm_management_group.level_1["/providers/Microsoft.Management/managementGroups/TestGroup"] will be created
  + resource "azurerm_management_group" "level_1" {
      + display_name               = "Test Group"
      + id                         = (known after apply)
      + name                       = "TestGroup"
      + parent_management_group_id = "/providers/Microsoft.Management/managementGroups/[...]"
      + subscription_ids           = (known after apply)
    }

  # module.enterprise_scale.time_sleep.after_azurerm_management_group will be created
  + resource "time_sleep" "after_azurerm_management_group" {
      + create_duration  = "30s"
      + destroy_duration = "0s"
      + id               = (known after apply)
      + triggers         = {
          + "azurerm_management_group_level_1" = jsonencode(
                [
                  + "/providers/Microsoft.Management/managementGroups/TestGroup",
                ]
            )
          + "azurerm_management_group_level_2" = jsonencode([])
          + "azurerm_management_group_level_3" = jsonencode([])
          + "azurerm_management_group_level_4" = jsonencode([])
          + "azurerm_management_group_level_5" = jsonencode([])
          + "azurerm_management_group_level_6" = jsonencode([])
        }
    }

  # module.enterprise_scale.time_sleep.after_azurerm_policy_assignment will be created
  + resource "time_sleep" "after_azurerm_policy_assignment" {
      + create_duration  = "30s"
      + destroy_duration = "0s"
      + id               = (known after apply)
      + triggers         = {
          + "azurerm_management_group_policy_assignment_enterprise_scale" = jsonencode([])
        }
    }

  # module.enterprise_scale.time_sleep.after_azurerm_policy_definition will be created
  + resource "time_sleep" "after_azurerm_policy_definition" {
      + create_duration  = "30s"
      + destroy_duration = "0s"
      + id               = (known after apply)
      + triggers         = {
          + "azurerm_policy_definition_enterprise_scale" = jsonencode([])
        }
    }

  # module.enterprise_scale.time_sleep.after_azurerm_policy_set_definition will be created
  + resource "time_sleep" "after_azurerm_policy_set_definition" {
      + create_duration  = "30s"
      + destroy_duration = "0s"
      + id               = (known after apply)
      + triggers         = {
          + "azurerm_policy_set_definition_enterprise_scale" = jsonencode([])
        }
    }

  # module.enterprise_scale.time_sleep.after_azurerm_role_assignment will be created
  + resource "time_sleep" "after_azurerm_role_assignment" {
      + create_duration  = "0s"
      + destroy_duration = "0s"
      + id               = (known after apply)
      + triggers         = {
          + "azurerm_role_assignment_enterprise_scale" = jsonencode([])
          + "module_role_assignments_for_policy"       = jsonencode([])
        }
    }

  # module.enterprise_scale.time_sleep.after_azurerm_role_definition will be created
  + resource "time_sleep" "after_azurerm_role_definition" {
      + create_duration  = "60s"
      + destroy_duration = "0s"
      + id               = (known after apply)
      + triggers         = {
          + "azurerm_role_definition_enterprise_scale" = jsonencode([])
        }
    }

Plan: 7 to add, 0 to change, 0 to destroy.

Terraform import module.enterprise_scale.azurerm_management_group.level_1 /providers/Microsoft.Management/managementGroups/TestGroup:
data.azurerm_client_config.core: Reading...
data.azurerm_client_config.core: Read complete after 0s [id=[...]]
╷
│ Error: Invalid for_each argument
│
│   on .terraform\modules\enterprise_scale\resources.management_groups.tf line 20, in resource "azurerm_management_group" "level_2":
│   20:   for_each = local.azurerm_management_group_level_2
│     ├────────────────
│     │ local.azurerm_management_group_level_2 will be known only after apply
│
│ The "for_each" map includes keys derived from resource attributes that cannot be determined until apply, and so
│ Terraform cannot determine the full set of keys that will identify the instances of this resource.
│
│ When working with unknown values in for_each, it's better to define the map keys statically in your configuration and
│ place apply-time results only in the map values.
│
│ Alternatively, you could use the -target planning option to first apply only the resources that the for_each value
│ depends on, and then apply a second time to fully converge.
```

**Comment by krowlandson** [comment]
@adhodgson1, please can you ensure you specify the correct value for `parent_management_group_id` to ensure it reflects the actual parent Management Group of the Management Group you want to import.

If this is the only MG you are managing in this module declaration, you will also need to set `root_parent_id` to the same value (which may well be the tenant ID).

**Comment by adhodgson1** [comment]
Thanks, specifying the parent_management_group_id fixed the issue.

**Comment by liamjvs** [comment]
If anyone stumbles on this same issue - resolved using Terraform v1.5 feature of an [`import`](https://developer.hashicorp.com/terraform/language/import) block.

```terraform
import {
  # Resource ID of the Management Group
  id = "/providers/Microsoft.Management/managementGroups/<Management Group ID>"
  # Terraform Resource Address of Management Group
  to = module.enterprise_scale.azurerm_management_group.level_1["/providers/Microsoft.Management/managementGroups/<Management Group ID>"]
}
```

Plan and Apply successful to a pre-existing Management Group:
```console
Plan: 1 to import, 208 to add, 0 to change, 0 to destroy.
```
```console
Apply complete! Resources: 1 imported, 208 added, 0 changed, 0 destroyed.
```

Thank you @matt-FFFFFF and @jtracey93
